File: src/main/java/com/cricket/cms/service/PlayerService.java
package com.cricket.cms.service;

import java.util.List;

import com.cricket.cms.dto.PlayerDTO;
import com.cricket.cms.entity.Player;

public interface PlayerService {

    Player addPlayer(PlayerDTO playerDTO);

    List<Player> getAllPlayers();

    Player getPlayerById(Long playerId);

    Player updatePlayer(Long playerId, PlayerDTO playerDTO);

    void deletePlayer(Long playerId);

}

File: frontend/src/services/PlayerService.js
import axios from "axios";

const BASE_URL = "http://localhost:8081/api/players";

class PlayerService {

    getAllPlayers() {
        return axios.get(BASE_URL);
    }

    getPlayerById(playerId) {
        return axios.get(BASE_URL + "/" + playerId);
    }

    addPlayer(player) {
        return axios.post(BASE_URL, player);
    }

    updatePlayer(playerId, player) {
        return axios.put(BASE_URL + "/" + playerId, player);
    }

}

export default new PlayerService();